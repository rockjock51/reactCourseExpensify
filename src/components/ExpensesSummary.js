import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

export const ExpensesSummary = props => (
  <div>
    <p>
      Showing {props.expenses.length} expenses totaling{" "}
      {numeral(props.expenseTotal(props.expenses) / 100).format("$0,0.00")}
    </p>
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: selectExpenses(state.expenses, state.filters),
    expenseTotal: selectExpensesTotal
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
