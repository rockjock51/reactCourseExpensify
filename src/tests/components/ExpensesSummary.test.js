import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import expenses from "../fixtures/expenses";

let expenseTotal;

beforeEach((expenseTotal = jest.fn()));

test("should render correctly", () => {
  const wrapper = shallow(
    <ExpensesSummary expenses={expenses} expenseTotal={expenseTotal} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render 1 expense correctly", () => {
  const wrapper = shallow(
    <ExpensesSummary expenses={expenses[1]} expenseTotal={expenseTotal} />
  );
  expect(wrapper).toMatchSnapshot();
});
