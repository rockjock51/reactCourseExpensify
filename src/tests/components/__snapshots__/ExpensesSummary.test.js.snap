// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1 expense correctly 1`] = `
<div>
  <p>
    Showing 
     expenses totaling
     
    $0.00
  </p>
</div>
`;

exports[`should render correctly 1`] = `
<div>
  <p>
    Showing 
    3
     expenses totaling
     
    $0.00
  </p>
</div>
`;
